library/azure_rm_cosmosdbaccount.py

```diff
@@ -0,0 +1,488 @@
+
+
+class AzureRMDatabaseAccounts(AzureRMModuleBase):
+    """Configuration class for an Azure RM Database Account resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            location=dict(
+                type='str'
+            ),
+            kind=dict(
+                type='str',
+                choices=['global_document_db',
+                         'mongo_db',
+                         'parse']
+            ),
+            consistency_policy=dict(
+                type='dict'
+            ),
+            locations=dict(
+                type='list'
+            ),
+            database_account_offer_type=dict(
+                type='str'
+            ),
+            ip_range_filter=dict(
+                type='str'
+            ),
+            is_virtual_network_filter_enabled=dict(
+                type='str'
+            ),
+            enable_automatic_failover=dict(
+                type='bool'
+            ),
+            capabilities=dict(
+                type='list'
+            ),
+            virtual_network_rules=dict(
+                type='list'
+            ),
+            enable_multiple_write_locations=dict(
+                type='str'
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.parameters = dict()
+
+        self.results = dict(changed=False)
+        self.mgmt_client = None
+        self.state = None
+        self.to_do = Actions.NoAction
+
+        super(AzureRMDatabaseAccounts, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                      supports_check_mode=True,
+                                                      supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            elif kwargs[key] is not None:
+                if key == "location":
+                    self.parameters["location"] = kwargs[key]
+                elif key == "kind":
+                    ev = kwargs[key]
+                    if ev == 'global_document_db':
+                        ev = 'GlobalDocumentDB'
+                    elif ev == 'mongo_db':
+                        ev = 'MongoDB'
+                    self.parameters["kind"] = _snake_to_camel(ev, True)
+                elif key == "consistency_policy":
+                    ev = kwargs[key]
+                    if 'default_consistency_level' in ev:
```

Comment:
pls use string.title() to upper first case. and string.lower() to lower. and helper utility snake_to_camel() to snake to camel. and vise